### PR TITLE
Complete Release notes

### PR DESCRIPTION
**Pull request by jrobinAV**
Added the release notes for the sdm, templates, config/common, enterprise topics

### CODE REVIEW

File: docs/release-notes/index.md [hunk 1]
```diff
@@ -33,6 +33,20 @@ This is the list of changes to Taipy releases as they were published.
 
 <h4><strong><code>taipy</code></strong> 4.0.0</h4>
 
+- The User Experience of the Scenario and Data management Visual elements have been greatly
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md:36]: Thanks for that!

File: docs/release-notes/index.md [hunk 2]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+- Support for different encodings in `S3ObjectDataNode^`.<br/>
+  See [issue #680](https://github.com/Avaiga/taipy/issues/680).
+- Reading an `ExcelDataNode^` is more consistent across the various expose types.<br/>
+  See [issue #796](https://github.com/Avaiga/taipy/issues/796).
 - Two scenarios belonging to the same cycle can now have the same tag.<br/>
-  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).
+  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).<br/>
+- The custom properties of a `Scenario` are not exposed as attribute anymore.<br/>
+  See [issue #1572](https://github.com/Avaiga/taipy/issues/1572).
+- Methods and functions returning booleans and related to entities now return
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
- Methods and functions returning a Boolean value and related to entities now return
```

File: docs/release-notes/index.md [hunk 3]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+  `ReasonCollection^` as a set of `Reason^` instances. Each reason contains a message
+  explaining why the boolean is `False`. The collection is empty if the boolean is `True`
+  .<br/>
+  Examples: `is_deletable^`, `exists^`, `is_readable^` etc.
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  Examples: `is_deletable()^`, `exists()^`, `is_readable()^` etc.
```

File: docs/release-notes/index.md [hunk 4]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+- Reading an `ExcelDataNode^` is more consistent across the various expose types.<br/>
+  See [issue #796](https://github.com/Avaiga/taipy/issues/796).
 - Two scenarios belonging to the same cycle can now have the same tag.<br/>
-  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).
+  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).<br/>
+- The custom properties of a `Scenario` are not exposed as attribute anymore.<br/>
+  See [issue #1572](https://github.com/Avaiga/taipy/issues/1572).
+- Methods and functions returning booleans and related to entities now return
+  `ReasonCollection^` as a set of `Reason^` instances. Each reason contains a message
+  explaining why the boolean is `False`. The collection is empty if the boolean is `True`
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  explaining why the boolean is `False`. The collection is empty if the Boolean is `True`
```

File: docs/release-notes/index.md [hunk 5]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+  See [issue #680](https://github.com/Avaiga/taipy/issues/680).
+- Reading an `ExcelDataNode^` is more consistent across the various expose types.<br/>
+  See [issue #796](https://github.com/Avaiga/taipy/issues/796).
 - Two scenarios belonging to the same cycle can now have the same tag.<br/>
-  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).
+  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).<br/>
+- The custom properties of a `Scenario` are not exposed as attribute anymore.<br/>
+  See [issue #1572](https://github.com/Avaiga/taipy/issues/1572).
+- Methods and functions returning booleans and related to entities now return
+  `ReasonCollection^` as a set of `Reason^` instances. Each reason contains a message
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  a `ReasonCollection^` object, which is a set of `Reason^` instances. Each reason contains a message
```

File: docs/release-notes/index.md (hunk 4 above)
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  explaining why the return value is False. The collection is empty if the returned value is True
```

File: docs/release-notes/index.md [hunk 6]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+  See [issue #1696](https://github.com/Avaiga/taipy/issues/1696) and
+  [issue #411](https://github.com/Avaiga/taipy/issues/411).<br/>
+  For more information on checkers, please refer to
   [Configuration checker](../userman/advanced_features/configuration/config-checker.md).
 
 <h4><strong><code>taipy-templates</code></strong> 4.0.0 </h4>
 
 - The *--template* option of the `taipy create` command is now renamed to *--application* option
-  to correctly reflect the application template to use when creating a new Taipy application.
+  to correctly reflect the application template to use when creating a new Taipy application. <br/>
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  to correctly reflect the application template to use when creating a new Taipy application.<br/>
```

File: docs/release-notes/index.md [hunk 7]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+  See[issue #1472](https://github.com/Avaiga/taipy/issues/1472)
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
  See [issue #1472](https://github.com/Avaiga/taipy/issues/1472)
```

File: docs/release-notes/index.md [hunk 8]
```diff
@@ -197,6 +245,13 @@ This is the list of changes to Taipy releases as they were published.
 
 - `DataNode.is_up_to_date()^` raises an error when the data node has never been written.<br/>
   See [issue #1198](https://github.com/Avaiga/taipy/issues/1198).
+- Reload the cache on all _build_manager() methods when the *repository_type* is changed.<br/>
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
- Reload the cache on all *_build_manager()* methods when the *repository_type* is changed.<br/>
```

File: docs/release-notes/index.md [hunk 9]
```diff
@@ -197,6 +245,13 @@ This is the list of changes to Taipy releases as they were published.
...
+  See [issue #1692](https://github.com/Avaiga/taipy/pull/1692)
+
+<h4><strong><code>taipy-templates</code></strong> 4.0.0</h4>
+- First cli option of the `taipy create` command is skipped if it's before the positional
```
Comment (FabienLelaquais) [inline docs/release-notes/index.md]:
```suggestion
- The first cli option of the `taipy create` command is skipped if it's before the positional
```

File: docs/release-notes/index.md (hunk 7 above)
Comment (FlorianJacta) [inline docs/release-notes/index.md]: Add a space between "See" and "["

**Review by FabienLelaquais — APPROVED** [review]
👌

File: docs/release-notes/index.md (hunk 7 above)
Comment (jrobinAV) [reply docs/release-notes/index.md]:
```suggestion
  See [issue #1472](https://github.com/Avaiga/taipy/issues/1472)
```

**Review by FlorianJacta — APPROVED** [review]
LGTM

File: docs/release-notes/index.md [hunk 10]
```diff
@@ -154,28 +180,50 @@ This is the list of changes to Taipy releases as they were published.
...
+- Reading an `ExcelDataNode^` is more consistent across the various expose types.<br/>
+  See [issue #796](https://github.com/Avaiga/taipy/issues/796).
 - Two scenarios belonging to the same cycle can now have the same tag.<br/>
-  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).
+  See [issue #1292](https://github.com/Avaiga/taipy/issues/1292).<br/>
+- The custom properties of a `Scenario` are not exposed as attribute anymore.<br/>
+  See [issue #1572](https://github.com/Avaiga/taipy/issues/1572).
+- Methods and functions returning a Boolean value and related to entities now return
+  a `ReasonCollection^` object, which is a set of `Reason^` instances. Each reason contains a message
+  explaining why the boolean is `False`. The collection is empty if the Boolean is `True`
```
Comment (jrobinAV) [inline docs/release-notes/index.md]:
```suggestion
  explaining why the returned value is `False`. The collection is empty if the value is `True`
```

**Pull request merged**